manifests/manifests.go

```diff
@@ -21,24 +21,50 @@ var (
 	wegoAppTemplates embed.FS
 )
 
-type WegoAppParams struct {
-	Version   string
-	Namespace string
+type Params struct {
+	AppVersion      string
+	ProfilesVersion string
+	Namespace       string
 }
 
-// GenerateWegoManifests generates wego-app manifests from a template
-func GenerateWegoAppManifests(params WegoAppParams) ([][]byte, error) {
-	manifests := [][]byte{}
+// GenerateManifests generates weave-gitops manifests from a template
+func GenerateManifests(params Params) ([][]byte, error) {
+	appManifests, err := readTemplateDirectory(params, wegoAppTemplates, wegoManifestsDir)
+	if err != nil {
+		return nil, fmt.Errorf("rendering app manifests: %w", err)
+	}
+
+	return appManifests, nil
+}
+
+func executeTemplate(name string, tplData string, params Params) ([]byte, error) {
+	template, err := template.New(name).Parse(tplData)
+	if err != nil {
+		return nil, fmt.Errorf("error parsing template %s: %w", name, err)
+	}
 
-	templates, err := fs.ReadDir(wegoAppTemplates, wegoManifestsDir)
+	yaml := &bytes.Buffer{}
+
+	err = template.Execute(yaml, params)
+	if err != nil {
+		return nil, fmt.Errorf("error injecting values to template: %w", err)
+	}
+
+	return yaml.Bytes(), nil
+}
+
+func readTemplateDirectory(params Params, templateFiles embed.FS, templatestDir string) ([][]byte, error) {
```

Comment:
`templatestDir` -> `templateDir` ?

Reply:
The name of the function is a bit misleading. It doesn't just read the template directory.

It gathers, executes and returns manifest data.

Can we call this `executeTemplates` or `renderManifestFiles`?

Reply:
renamed